timeseries/src/autogluon/timeseries/transforms/covariate_scaler.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from typing import Dict, List, Literal, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import QuantileTransformer, StandardScaler
+
+from autogluon.timeseries.dataset.ts_dataframe import TimeSeriesDataFrame
+from autogluon.timeseries.utils.features import CovariateMetadata
+from autogluon.timeseries.utils.warning_filters import warning_filter
+
+logger = logging.getLogger(__name__)
+
+
+class CovariateScaler:
+    """Preprocesses covariates and static features."""
+
+    def __init__(
+        self,
+        metadata: CovariateMetadata,
+        use_known_covariates: bool = True,
+        use_past_covariates: bool = True,
+        use_static_features: bool = True,
+        **kwargs,
+    ):
+        self.metadata = metadata
+        self.use_known_covariates = use_known_covariates
+        self.use_past_covariates = use_past_covariates
+        self.use_static_features = use_static_features
+
+    def fit_transform(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: Optional[TimeSeriesDataFrame] = None,
+    ) -> Tuple[TimeSeriesDataFrame, Optional[TimeSeriesDataFrame]]:
```

Comment:
The fact that `LocalTargetScaler.transform() -> TimeSeriesDataFrame` and `CovariateScaler.transform() -> Tuple` is not trivial. It might be worth documenting this in the transform methods of both classes.

Reply:
Good point, will add a docstring. The alternative to returning a tuple is saving an internal state and applying to `known_covariates` - which is a bit more tedious

Reply:
You are right, we can actually split the preprocessing for data & known_covariates into separate methods, making the API cleaner.